### PR TITLE
You can now control-click action buttons to lock them in place

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
tweak: You can now control-click action buttons to lock them and prevent them from being moved. Alt-clicking the "Show/Hide Actions" button will unlock all buttons.
tweak: There is now a preference for if buttons should be locked by default or not.
/:cl:

This is a thing that felt strangely lacking, and I was reminded of it by some chatter in coderbus, so here it is. Ctrl-clicking action buttons now locks them in place and prevents them from being moved until you unlock them. Resetting the positions of all action buttons (through alt-clicking the Show/Hide Actions button will undo all locks.)

### CODE REVIEW

File: code/_onclick/hud/movable_screen_objects.dm [hunk 1]
```diff
@@ -20,6 +21,8 @@
 
 
 /obj/screen/movable/MouseDrop(over_object, src_location, over_location, src_control, over_control, params)
+	if(locked) //no! I am locked! begone!
+		return
```
Comment (Cyberboss) [inline code/_onclick/hud/movable_screen_objects.dm]: Does this actually prevent the dragging or only the drop?

Comment (RemieRichards) [inline code/_onclick/hud/movable_screen_objects.dm]: yes, it's the MouseDrop that 'commits' the movement.

**Comment by WJohn** [comment]
These should start locked tbh. I never want to drag them in the first place.

**Comment by Cyberboss** [comment]
Should be a preference tbh

**Comment by Xhuis** [comment]
What, starting locked?

**Comment by Cyberboss** [comment]
Yes

**Comment by Xhuis** [comment]
It seems a little small to implement a pref for. I'll do it, but do you feel like it's worth it?

**Comment by Armhulen** [comment]
yessssssssssssss

**Comment by Cyberboss** [comment]
please do

**Comment by Xhuis** [comment]
Preference added! It defaults to unlocked. Action buttons are, by default, locked or unlocked, depending on what that preference is set to.

**Comment by optimumtact** [comment]
it would help if it compiles

**Comment by Xhuis** [comment]
You can't control me

File: code/_onclick/hud/movable_screen_objects.dm (hunk 1 above)
Comment (optimumtact) [inline code/_onclick/hud/movable_screen_objects.dm]: I think cyberboss was trying to ask if this disabled the dragging or just made it snap back into position after you finished dragging it.

Comment (Xhuis) [inline code/_onclick/hud/movable_screen_objects.dm]: The position changes in `MouseDrop`, so this disables the moving.

**Pull request merged**